ocs_ci/utility/baremetal.py

```diff
@@ -0,0 +1,207 @@
+            wait (bool): Wait for BMs to start
+
+        """
+        for ipmi_ctx in ipmi_ctxs:
+            ipmi_ctx.chassis_control_power_up()
+
+        if wait:
+            for ipmi_ctx in ipmi_ctxs:
+                for status in TimeoutSampler(600, 5, self.get_power_status, ipmi_ctx):
+                    logger.info(
+                        f"Waiting for Baremetal Machine to power on. "
+                        f"Current Baremetal status: {status}"
+                    )
+                    if status == VM_POWERED_ON:
+                        logger.info("Baremetal Machine reached poweredOn status")
+                        break
+
+        wait_for_cluster_connectivity(delay=5)
+        wait_for_nodes_status(
+            node_names=helpers.get_master_nodes(),
+            status=constants.NODE_READY
+        )
+        wait_for_nodes_status(
+            node_names=helpers.get_worker_nodes(),
+            status=constants.NODE_READY
+        )
+
+    def start_baremetal_machines(self, baremetal_machine, wait=True):
+        """
+        Start Baremetal Machines
+
+        Args:
+            baremetal_machine (list): BM objects
+            wait (bool): Wait for BMs to start
+
+        """
+        for node in baremetal_machine:
+            for details in self.mgmt_details:
+                if node.name == details['hostname']:
+                    ipmi_ctx = self.get_ipmi_ctx(
+                        host=details['mgmt_console'], user=details['mgmt_username'],
+                        password=details['mgmt_password']
+                    )
+                    logger.info(f"Powering On {node.name}")
+                    ipmi_ctx.chassis_control_power_up()
+            if wait:
+                for details in self.mgmt_details:
+                    if node.name == details['hostname']:
+                        ipmi_ctx = self.get_ipmi_ctx(
+                            host=details['mgmt_console'], user=details['mgmt_username'],
+                            password=details['mgmt_password']
+                        )
+                        for status in TimeoutSampler(600, 5, self.get_power_status, ipmi_ctx):
+                            logger.info(
+                                f"Waiting for Baremetal Machine {node.name} to power on. "
+                                f"Current Baremetal status: {status}"
+                            )
+                            if status == VM_POWERED_ON:
+                                logger.info(f"Baremetal Machine {node.name} reached poweredOn status")
+                                ipmi_ctx.session.close()
+                                break
+
+        wait_for_cluster_connectivity(delay=5)
+        wait_for_nodes_status(
+            node_names=helpers.get_master_nodes(),
+            status=constants.NODE_READY
+        )
+        wait_for_nodes_status(
+            node_names=helpers.get_worker_nodes(),
+            status=constants.NODE_READY
+        )
```

Comment:
Do we really need to wait for all the master and worker nodes? Won't it cause a test to fail if it is not expecting all the nodes to be Ready?

Don't want to block this PR if this requires major change, so this can be taken up in follow up PR.